### ISSUE TITLE
Auth/EPP Keys Too long

### ISSUE THREAD

**Issue by whitetiger264** [getnamingo/whmcs-epp-registrar#13]
When requesting an Auth Code/EPP Key for transfer, I am presented with the following text `dec8da9c7bd40e3cae51d7b2bf930ea3e19db55045304160a9aae69cf5fa11f7353526153691e47fc39868d705c32466a6b4569642bbd6b6bbxxxxxxxxxxxxxxxxxxx`  This is way too long. Can you look at shortening it?

**Comment by getpinga** [comment]
This is something with your registry, as currently we generate much shorter. Did you find a fix? Or let me know what is the registry you connect to. Thank you.

**Comment by whitetiger264** [comment]
I am using another WHMCS module, and the EPP keys we get are short, I do not think it is the registry in this case. The registry we're using uses the CoCCA system.

**Comment by getpinga** [comment]
I'll try to get Cocca instance to test, I think they don't use anymore the authcode like before, they generate the password on request. And maybe it is done in the module which itself sets temporarily the password. What my module pulls from EPP is some hash which is useless.

**Comment by getpinga** [comment]
Can you please tell me which module you are using? It seems for Cocca must be done domain update with new random auth code and then show this code.

**Comment by whitetiger264** [comment]
A WHMCS module that Cocca released or provided a few years back. I am not sure if I can share it, as I have no idea what legal implications it may have.

**Comment by getpinga** [comment]
I know about it, but it seems to have disappeared from the Internet. So don't share it, the one I found from them is what I mentioned - generating random authcode, updating the domain and then displaying it. Will see how to integrate it.

**Comment by whitetiger264** [comment]
Yep, should be that one. It seems it's just MIA, which is why I am hesitant to send it. It does, however, seem that @dnsguru was the last to work on it in 2020. Maybe we can get a comment on this.

**Comment by dnsguru** [comment]
> Yep, should be that one. It seems it's just MIA, which is why I am hesitant to send it. It does, however, seem that @dnsguru was the last to work on it in 2020. Maybe we can get a comment on this. 

I did a quick refresh of the documentation on the whmcs module back in 2020 to catch it up.

Now that I look, it appears CoCCA are not listing it so it may be something they include upon request.  Not much help, but there is your comment

**Comment by dnsguru** [comment]
They operate on wafer-thin income and depend on their licensing to help drive development and support.

I would recommend contacting CoCCA directly with the contact you work with to license and consider using the most recent patch.

**Comment by whitetiger264** [comment]
Thank you @dnsguru the feedback is appreciated.

**Comment by getpinga** [comment]
Thank you, both. I found the needed explanation https://cocca.org.nz/srs/faq.html

**Comment by getpinga** [comment]
Finally after some time made a separate version https://github.com/getnamingo/whmcs-epp-cocca that would work with the CoCCA authcodes well. Closing the ticket, please open an issue there if you have one. Thank you for your patience.

**Comment by whitetiger264** [comment]
@getpinga is there any way you can update this repo with this EPP code fix, please? We currently use the CoCCA WHMCS module already and also rely on your generic epp module for other use cases.

**Comment by getpinga** [comment]
@whitetiger264 Still testing a few options as from experience not all registries support domain update for authcode  and the old way of domain info is still valid for most but Cocca.
